File: AvernusGames_Back-End/src/Models/Vestimenta.cs
using Microsoft.EntityFrameworkCore;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Avernus_Games_Store.src.Models
{
    public class Vestimenta : Produto
    {
        public Vestimenta(){} // Construtor vazio
        public override float CalcValorVenda(float ValorCompra, float Markup)
        {
             float ValorVenda = ValorCompra * (1 + Markup);
            float taxaImposto = 0.19f; // 19% de imposto
            float ValorTotal = ValorVenda * (1 + taxaImposto);
            return ValorTotal;
        }
    }
}

File: AvernusGames_Front-End/src/app/models/Vestimenta.ts
import { Produto } from "./Produto";

export class Vestimenta extends Produto {
    
    public CalcValorVenda(ValorCompra: number, Markup: number): number {
        const ValorVenda: number = ValorCompra * (1 + Markup);
        const taxaImposto: number = 0.19; // 19% de imposto
        const ValorTotal: number = ValorVenda * (1 + taxaImposto);
        return ValorTotal;
      }

}